packages/react-app/pages/index.tsx

```diff
@@ -2,23 +2,36 @@ import { useEffect, useState } from "react";
 import { useAccount } from "wagmi";
 
 export default function Home() {
-  const [userAddress, setUserAddress] = useState("");
-  const { address, isConnected } = useAccount();
+    const [userAddress, setUserAddress] = useState("");
+    const [isMounted, setIsMounted] = useState(false);
+    const { address, isConnected } = useAccount();
 
-  useEffect(() => {
-    if (isConnected && address) {
-      setUserAddress(address);
+    useEffect(() => {
+        setIsMounted(true);
+    }, []);
+
+    useEffect(() => {
+        if (isConnected && address) {
+            setUserAddress(address);
+        }
+    }, [address, isConnected]);
+
+    if (!isMounted) {
+        return null;
     }
-  }, [address, isConnected]);
 
-  return (
-    <div className="flex flex-col justify-center items-center">
-      <div className="h1">
-        There you go... a canvas for your next Celo project!
-      </div>
-      {isConnected && (
-        <div className="h2 text-center">Your address: {userAddress}</div>
-      )}
-    </div>
-  );
+    return (
+        <div className="flex flex-col justify-center items-center">
+            <div className="h1">
+                There you go... a canvas for your next Celo project!
```

Comment:
We talked about maybe having a similar setup as the minipay template here so people already have a read and a write function, that they can just copy. Could be something as easy as readign their cUSD balance and sending cUSD

Reply:
Yeah that can be done, will add that in separate PR.

But did you test celo-composer and did it work for you (without any errors and having to change anything)?

Reply:
Separate PR for that coming soon...